cmd/iam.go

```diff
@@ -352,42 +355,40 @@ func (sys *IAMSys) LoadUser(objAPI ObjectLayer, accessKey string, userType IAMUs
 }
 
 // Load - loads iam subsystem
-func (sys *IAMSys) Load() error {
+func (sys *IAMSys) Load(ctx context.Context) error {
 	// Pass nil objectlayer here - it will be loaded internally
 	// from the IAMStorageAPI.
-	return sys.store.loadAll(sys, nil)
+	return sys.store.loadAll(ctx, sys)
 }
 
 // Perform IAM configuration migration.
-func (sys *IAMSys) doIAMConfigMigration(objAPI ObjectLayer) error {
-	return sys.store.migrateBackendFormat(objAPI)
+func (sys *IAMSys) doIAMConfigMigration(ctx context.Context) error {
+	return sys.store.migrateBackendFormat(ctx)
 }
 
 // Init - initializes config system from iam.json
-func (sys *IAMSys) Init(objAPI ObjectLayer) error {
+func (sys *IAMSys) Init(ctx context.Context, objAPI ObjectLayer) error {
 	if objAPI == nil {
 		return errServerNotInitialized
 	}
 
-	if globalLDAPConfig.Enabled {
-		sys.EnableLDAPSys()
-	}
-
-	sys.Lock()
 	if globalEtcdClient == nil {
-		sys.store = newIAMObjectStore()
+		sys.store = newIAMObjectStore(ctx, objAPI)
 	} else {
-		sys.store = newIAMEtcdStore()
+		sys.store = newIAMEtcdStore(ctx)
+	}
+
+	if globalLDAPConfig.Enabled {
+		sys.EnableLDAPSys()
 	}
-	sys.Unlock()
 
 	// Migrate IAM configuration
-	if err := sys.doIAMConfigMigration(objAPI); err != nil {
+	if err := sys.doIAMConfigMigration(ctx); err != nil {
```

Comment:
Maybe we should not use the global context here so the server won't be killed during migration

Reply:
It is on purpose, we can't hang forever @vadmeste